### PR TITLE
Feat: Add annotations to runtime PODs

### PR DESCRIPTION
**Pull request by manuellysuzik**
Create annotation into pod metadata

### CODE REVIEW

File: internal/adapters/runtime/kubernetes/game_room_convert_test.go [hunk 1]
```diff
@@ -373,6 +374,7 @@ func TestConvertGameSpec(t *testing.T) {
 	cases := map[string]struct {
 		schedulerID string
 		roomName    string
+		Annotations map[string]string
```
Comment (reinaldooli) [inline internal/adapters/runtime/kubernetes/game_room_convert_test.go:377]: nit: this can be made private

File: internal/core/ports/runtime.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ type Runtime interface {
 	DeleteScheduler(ctx context.Context, scheduler *entities.Scheduler) error
 	// CreateGameRoomInstance Creates a game room instance on the runtime using
 	// the specification provided.
-	CreateGameRoomInstance(ctx context.Context, schedulerId, gameRoomName string, spec game_room.Spec) (*game_room.Instance, error)
+	CreateGameRoomInstance(ctx context.Context, scheduler entities.Scheduler, gameRoomName string, spec game_room.Spec) (*game_room.Instance, error)
```
Comment (reinaldooli) [inline internal/core/ports/runtime.go:41]: I think it's better to use the reference/pointer of the scheduler here.

Comment (manuellysuzik) [reply internal/core/ports/runtime.go:41]: sure

**Review by tobiasbu — APPROVED** [review]
LGTM!

**Review by wl-yankabuki — APPROVED** [review]

**Pull request merged**